### ISSUE TITLE
PodamUtils.getIntegerRange() produces non-uniformly distributed values

### ISSUE THREAD

**Issue by Otanikotani** [devopsfolks/podam#236]
Code, it simply counts frequencies of random values from 0 to 2 inclusively. 
```java
    @Test
    public void test_() throws Exception {
        Map<Integer, Integer> counters = new HashMap<>();
        counters.put(0, 0);
        counters.put(1, 0);
        counters.put(2, 0);
        for (int i = 0; i < 10_000_000; i++) {
            int result = PodamUtils.getIntegerInRange(0, 2);
            Integer counter = counters.get(result) + 1;
            counters.put(result, counter);
        }
        System.out.println(counters);
    }
```

Result:
```
{0=2500466, 1=5000374, 2=2499160}
```

Is it expected behaviour? For range from 0 to 10 same code generates:
```
{0=500043, 1=999784, 2=1001485, 3=1001217, 4=998733, 5=999781, 6=1000986, 7=999542, 8=998796, 9=999932, 10=499701}
```

Which looks a little bit weird and not fair!

**Comment by daivanov** [comment]
Yes, this is an effect of rounding nobody was paying attention to earlier. Not sure what is expected behaviour here, but I could imagine it might affect testing of marginal values.

**Comment by daivanov** [comment]
Snapshot [7.0.5-2](https://oss.sonatype.org/content/repositories/snapshots/uk/co/jemos/podam/podam/7.0.5-SNAPSHOT/podam-7.0.5-20170205.182138-2.jar).

**Comment by Otanikotani** [comment]
@daivanov May I ask you why implementation

```java
new Random().nextInt((max - min) + 1) + min;
```

Is not used? Why is it important to add and substract 0.5?

**Comment by daivanov** [comment]
And why should it be used? ;)

**Comment by Otanikotani** [comment]
Well, did you always create a new instance of Random?

**Comment by daivanov** [comment]
Sure, because it was part of the question.

**Comment by Otanikotani** [comment]
Benchmark:

```java
public class Randoms {

    @Benchmark
    public Integer current() {
        return PodamUtils.getIntegerInRange(0, 1000);
    }

    @Benchmark
    public Integer rangeFromSo() {
        return PodamUtils.getIntegerRangeDefaultVersionFromStackOverflow(0, 1000);
    }

    public static void main(String[] args) throws RunnerException {
        Options opt = new OptionsBuilder()
                .include(Randoms.class.getSimpleName())
                .forks(1)
                .build();

        new Runner(opt).run();
    }
}
```

Results:
```
Benchmark             Mode  Cnt         Score        Error  Units
Randoms.current      thrpt   20  35493740.187 ± 304428.342  ops/s
Randoms.rangeFromSo  thrpt   20  75697058.949 ± 554676.207  ops/s
```

Where methods are like this:

```java
    public static int getIntegerInRange(int minValue, int maxValue) {

        return (int) (getDoubleInRange(minValue - 0.5, maxValue + 0.5 - (1 / Integer.MAX_VALUE)) + 0.5);
    }

    public static int getIntegerRangeDefaultVersionFromStackOverflow(int minValue, int maxValue) {
        return RNG.nextInt((maxValue - minValue) + 1) + minValue;
    }
```

and RNG is static final variable.

**Comment by Otanikotani** [comment]
But why would you measure performance for code which is not optimized? Of course it would lose if you create an instance every time. The point of that version was to show that it is simply more readable without casts, division and strange values like 0.5.

**Comment by daivanov** [comment]
And can you do this for long?

**Comment by Otanikotani** [comment]
It is possible to do without implementing your own version in Java 7 or using apache commons math.

**Comment by daivanov** [comment]
But you also should tell why things should be done. People should know motivational part of the proposal.

**Comment by daivanov** [comment]
I will resolve this issue, if you don't mind.

**Comment by Otanikotani** [comment]
Yea, sure, it is all only speculative, I don't think that normalized distribution or performance of random generator is that important anyway.

**Comment by daivanov** [comment]
The distribution is not normal, now it is uniform. Performance is of course important, but it's not easy to improve it.

**Comment by daivanov** [comment]
Released in Podam 7.0.5
